### PR TITLE
[CSLD-23] Transaction semantic description

### PR DESCRIPTION
**Pull request by denisshevchenko**

### CODE REVIEW

File: _docs/cardano/2017-01-04-transactions.md [hunk 1]
```diff
@@ -3,41 +3,113 @@ layout: default
...
+1. An id of transaction `TxN` which output is used for this input. Transaction's id is a BLAKE2s-256 hash of the transaction, something like `f9bcbe752aee4512457f1fd350200cf870906b7e6e836688c9a3779645c86c01`.
+2. An index of the using output in `TxN`'s outputs. This index is 32-bit unsigned integer.
+
+And each output contains:
 
-Transactions are signed with the issuer's secret key. All of this,
-including selection of unspent outputs to be used as inputs, is done
-automatically, and you don't have to worry about it when making a
-transaction using Daedalus UI.
+1. An address of the node `N` we want to send a value to. An address is a BLAKE2s-224 hash of the public key of the `N` node, something like `88ca7f79d4edcf911b60eeb96c8e9284d0c07c6e61c59a9e1c17a5e9`.
```
Comment (gromakovsky) [inline _docs/cardano/2017-01-04-transactions.md]: Address is hash of hash of public key. It's described here: https://github.com/input-output-hk/cardano-docs.iohk.io/blob/csld23-transaction-semantic/_docs/cardano/2017-01-02-addresses.md#hashing

**Comment by gromakovsky** [comment]
Nothing is written about transaction verification which involves something scripts-related. I am not sure if it is fine or not.

File: _docs/cardano/2017-01-04-transactions.md [hunk 2]
```diff
@@ -3,41 +3,113 @@ layout: default
...
+------------------------+           +----------------------------------------------+                                             
+~~~
+
+Transaction `f9bcbe752aee4512..` informs us that:
+
+1. we want to send 100 ADA to the node with address `88ca7f79..`;
+2. this money corresponds to `0`-th output of the previous transaction with an id `891971a4c..`.
+
+So in this case we know that 100 ADA transfers from the current node (who generates transaction `f9bcbe752aee4512..`)
+to the node `88ca7f79..`, and this money was received (in some past) from the node `a00e4bb2..`.
```
Comment (denisshevchenko) [inline _docs/cardano/2017-01-04-transactions.md]: Will be fixed.

**Comment by denisshevchenko** [comment]
> Nothing is written about transaction verification which involves something scripts-related. I am not sure if it is fine or not.

Ok, I think it'd be useful to add such an info, thanks.

File: _docs/cardano/2017-01-04-transactions.md [hunk 3]
```diff
@@ -3,41 +3,113 @@ layout: default
...
+  |                      |      |                      |
+  |  Inputs     Outputs  |      |  Inputs     Outputs  |
+  | +------+   +-------+ |      | +------+   +-------+ |
+  | | In0  |   | Out0  + |      | | In0  |   | Out0  | |
+  | +------+   +-------+ |      | +------+   +-------+ |
+  | | In1  |   | Out1  | |      | | In1  |   | Out1  | |
+  | +------+   +-------+ |      | +------+   +-------+ |
+  | | ...  |   | ...   | |      | | ..   |   | ...   | |
+  | +------+   +-------+ |      | +------+   +-------+ |
+  | | InN  |   | OutN  | |      | | InN  |   | OutN  | |
```
Comment (ChShersh) [inline _docs/cardano/2017-01-04-transactions.md]: I think it's better to name last output as `OutM` to show that number of inputs and outputs can be different.

File: _docs/cardano/2017-01-04-transactions.md [hunk 4]
```diff
@@ -3,41 +3,113 @@ layout: default
...
+2. An index of the using output in `TxN`'s outputs. This index is 32-bit unsigned integer.
+
+And each output contains:
 
-Transactions are signed with the issuer's secret key. All of this,
-including selection of unspent outputs to be used as inputs, is done
-automatically, and you don't have to worry about it when making a
-transaction using Daedalus UI.
+1. An address of the node `N` we want to send a value to. An address is a BLAKE2s-224 hash of the hash of the public key of the `N` node, something like `88ca7f79d4edcf911b60eeb96c8e9284d0c07c6e61c59a9e1c17a5e9`. Please read about [Addresses in Cardano SL](/cardano/addresses/) for more info.
+2. Amount of  we want to send. This value is 64-bit unsigned integer with [hardcoded](https://github.com/input-output-hk/cardano-sl/blob/63adb31e813e21ec9da21cfa69984840308bbfa2/src/Pos/Types/Core.hs#L88) maximum value.
```
Comment (ChShersh) [inline _docs/cardano/2017-01-04-transactions.md]: Is this word after `Amount of ...` intentionally missed?

File: _docs/cardano/2017-01-04-transactions.md [hunk 5]
```diff
@@ -3,41 +3,113 @@ layout: default
...
-full node, you might want to delete old proofs after some time in order
-to save space.
+[witness](https://github.com/input-output-hk/cardano-sl/blob/63adb31e813e21ec9da21cfa69984840308bbfa2/src/Pos/Types/Types.hs#L93)) that this transaction is legitimate. Witness for transaction `TxN` consists of the witnesses of
+all `TxN`'s inputs. Witnesses are stored in the blockchain and anybody can see, inspect 
+and independently verify them. But after some time node can delete old proofs in order to save space.
 
 The technique of storing transactions separately from their proofs is
-called “segregated witness” (you may have heard of it being recently
+called "segregated witness" (you may have heard of it being recently
 implemented in Bitcoin). Under this scheme, transactions and proofs are
```
Comment (ChShersh) [inline _docs/cardano/2017-01-04-transactions.md]: Maybe link to Bitcoin implementation/news/announce would be nice here.

Comment (neongreen) [inline _docs/cardano/2017-01-04-transactions.md]: @denisshevchenko Why would you change right quotes to wrong ones? (I don't care much, but still curious.)

File: _docs/cardano/2017-01-04-transactions.md [hunk 6]
```diff
@@ -3,41 +3,113 @@ layout: default
...
-a list of inputs and a list of outputs. Outputs of a transaction can later
-be used as inputs for another transaction.
+## Overview
 
-Every node (except for SPV, or lightweight nodes) in the network
-verifies transactions, so those nodes have to keep track of unspent
-outputs, this is called “utxo”, or “Unspent Transaction Outputs”. By
-tracking utxo, every node can validate that inputs in a published
-transaction are indeed the unspent outputs.
+A transaction (_tx_) is a special data represents the _act_ of the value transferring
```
Comment (neongreen) [inline _docs/cardano/2017-01-04-transactions.md]: Not quite between two nodes:

* several nodes can have access to the same private key
* money can be transferred to an address that doesn't correspond to any existing private key

File: _docs/cardano/2017-01-04-transactions.md [hunk 7]
```diff
@@ -3,41 +3,119 @@ layout: default
...
+  +----------------------+      +----------------------+     ...
+~~~
+
+Inputs and outputs inform us about _money flow_: input informs where the money came from, and output informs where the money come to.
+Please notice that there's `N` and `M`, because actual number of inputs and outputs can be different.
+
+So, each input contains:
+
+1. An id of transaction `TxN` which output is used for this input. Transaction's id is a BLAKE2s-256 hash of the transaction, something like `f9bcbe752aee4512457f1fd350200cf870906b7e6e836688c9a3779645c86c01`.
+2. An index of the using output in `TxN`'s outputs. This index is 32-bit unsigned integer.
```
Comment (neongreen) [inline _docs/cardano/2017-01-04-transactions.md]: Actually a varint, though I'm not sure it should be mentioned here.

Comment (ChShersh) [inline _docs/cardano/2017-01-04-transactions.md]: Yes, it is indeed varint. Probably should be `TinyVarInt`.

**Comment by manpages** [comment]
@denisshevchenko please, when you think you've addressed all the concerns, ask @gromakovsky to review. Please do it today because he's on documentation _fully_ only today.

**Pull request merged**